connectors/src/connectors/intercom/temporal/sync_conversation.ts

```diff
@@ -87,21 +88,21 @@ export async function deleteConversation({
   ]);
 }
 
-export async function syncConversation({
+export async function fetchAndSyncConversation({
```

Comment:
On this file I renamed `syncConversation` into `fetchAndSyncConversation` so that the `syncConversation` that takes directly the intercom `IntercomConversationWithPartsType` object can shared here and from the webhook. 

+ Introduced a  `syncType` so that we make the different with the batch update and incremental one.